hasher-matcher-actioner/hmalib/lambdas/fetcher.py

```diff
@@ -333,10 +348,40 @@ def _apply_updates_impl(self, delta: tu.ThreatUpdatesDelta) -> None:
                 state.pop(item.key, None)
             else:
                 state[item.key] = item
+                updated[item.key] = item
 
         self._store_state(state.values())
         self._cached_state = state
 
+        post_apply_fn(updated)
+
+    def post_apply(self, updated: t.Dict = {}):
+        """
+        After the fetcher applies an update, check to see if
+        matches for any of the updated signals and if so update
+        their tags.
+        """
+        table = dynamodb.Table(self.data_store_table)
+
+        for update in updated.values():
+            row = update.as_csv_row()
+            # example row format: ('<signal>', '<id>', '<time added>', '<tag1 tags2>')
+            # e.g ('096a6f9...064f', 1234567891234567, '2020-07-31T18:47:45+0000', 'true_positive hma_test')
+            if PDQSignalMetadata(
+                signal_id=int(row[1]),
+                ds_id=str(self.privacy_group),
+                updated_at=datetime.now(),
+                signal_source=S3ThreatDataConfig.SOURCE_STR,
+                signal_hash=row[0],  # note: not used by update_tags_in_table_if_exists
+                tags=row[3].split(" ") if row[3] else [],
+            ).update_tags_in_table_if_exists(table):
```

Comment:
Why do we only update tags? is there an easy way to all the fields (except signal id and hash)

Reply:
The only meaningful field that isn't a part of lookup keys are tags and hash.

Reply:
Do/should we change the updated_at field also? If so, maybe we should change the name to update_pdq_metadata.

non blocking

Reply:
we already do, because 'update_at' is (or at least imo should be) updated every time an object in the db is updated (this field also exists for Match and Hash Records) and is not about the signal itself.